packages/next-server/server/next-server.ts

```diff
@@ -226,6 +234,32 @@ export default class Server {
     return routes
   }
 
+  /**
+   * Resolves `API` request, in development builds on demand
+   * @param req http request
+   * @param res http response
+   * @param pathname path of request
+   */
+  private async handleApiRequest(req: IncomingMessage, res: ServerResponse, pathname: string) {
+    const resolverFunction = await this.resolveApiRequest(pathname)
+    if (resolverFunction === null) {
+      res.statusCode = 501
+      res.end('Not Implemented')
+      return
+    }
+
+    const resolver = requireResolver(resolverFunction)
```

Comment:
Instead of adding a new function you could use this one https://github.com/zeit/next.js/blob/c9d599b69864f4a6c08cada4d0c9733cc630ddc4/packages/next-server/server/load-components.ts#L6